server.js

```diff
@@ -1,35 +1,125 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb+srv://tuttibalutti:pOSlRc4H7PATpf2n@cluster0.c3qao.mongodb.net/bookFinder?retryWrites=true&w=majority"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+//Book collection
+const bookSchema = new mongoose.Schema({
+  title: String,
+  author: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Author'  
+  },
+})
+
+const Book = mongoose.model('Book', bookSchema)
+
+//Author collection
+const authorSchema = new mongoose.Schema({
+  name: String
+})
+
+const Author = mongoose.model('Author', authorSchema)
+
+//Seed the database
+if (process.env.RESET_DB) { 
+  const seedDB = async () => {
+    await Book.deleteMany()
+    await Author.deleteMany()
+
+    let authorsNames = []
+
+    booksData.forEach(book => {
+      book.authors.split('-').forEach(author => {
+        authorsNames.push(author)
+      })
+    })
+
+    authorsNames = [...new Set(authorsNames)]
+
+    let authorList = []
+
+    authorsNames.forEach(async authorName => {
+      const newAuthor = new Author({ name: authorName })
+      authorList.push(newAuthor)
+      await newAuthor.save()
+    })
+
+    booksData.forEach(async book => {
+      const newBook = new Book({
+        ...book,
+        author: authorList.find(singleAuthor => book.authors.includes(singleAuthor.name))
+      })
+      await newBook.save()
+    })
+   
+  }
+seedDB()
+}
+
+// Port where the app runs.
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// Middleware to enable cors
 app.use(cors())
-app.use(bodyParser.json())
 
-// Start defining your routes here
+// Routes
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Welcome to your bookFinder API')
+})
+
+app.get('/books', async (req, res) => {
+  const books = await Book.find()
+  res.json(books)
+})
+
+app.get('/books/title', async (req, res) => {
```

Comment:
When you use a query to filter your data, you need to use it on the list endpoint:
`/books?title=harry` 
not 
`/books/title?title=harry`